flocker/node/agents/ebs.py

```diff
@@ -590,6 +602,11 @@ def destroy_volume(self, blockdevice_id):
                         'Failed to delete volume: {!r}'.format(blockdevice_id)
                     )
                 else:
+                    ebs_volume = self._get_ebs_volume(blockdevice_id)
+                    _wait_for_volume(ebs_volume,
+                                     start_status=u'available',
+                                     transient_status=u'deleting',
+                                     end_status='')
```

Comment:
I wonder why `destroy_volume` uses `self.list_volumes()` instead of using `self._get_ebs_volume` right from the start.